### PR TITLE
Zerofill

### PR DESCRIPTION
**Pull request by joaomcteixeira**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and you comply with the following criteria:

- [x] You have stick to Python. Talk with us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [x] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [x] You structured the code into small functions as much as possible. You can use classes if there's a (state) purpose
- [x] code follows our coding style
- [x] You wrote tests for the new code
- [x] `tox` tests pass. *Run `tox` command inside the repository folder*
- [x] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [x] PR does not add any *install dependencies* unless permission granted by the HADDOCK team
- [x] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Modules folders are now zerofilled according to to the number of modules. If there are less than 10 modules, module folder will be prefix with only one digit. More than 10 modules, would be prefixed with 0, and so on. Thanks @douweschulte for the input :wink:

### CODE REVIEW

File: src/haddock/libs/libutil.py [hunk 1]
```diff
@@ -281,6 +281,16 @@ def get_number_from_path_stem(path):
...
 
 
+def get_number_of_digits(num):
+    """
+    Get the number of digits of a number.
+
+    10 has two digits.
+    100 has three digits.
+    """
+    return len(str(num))
```
Comment (douweschulte) [inline src/haddock/libs/libutil.py:291]: `log10(num+1)` also gives this result, and 'should' be faster (although that does not matter here). But I feel like a numerical problem should not be solved by using strings ;-)

File: src/haddock/modules/__init__.py [hunk 2]
```diff
@@ -198,7 +198,8 @@ def _load_previous_io(self):
 
     def previous_path(self):
         """Give the path from the previous calculation."""
-        previous = sorted(list(self.path.resolve().parent.glob('[0-9][0-9]*/')))
+        # https://regex101.com/r/4qRTjn/1
+        previous = sorted(list(self.path.resolve().parent.glob('[0-9]*/')))
```
Comment (douweschulte) [inline src/haddock/modules/__init__.py:202]: Why does this need to be changed? It already parsed one or more digits which is what is needed. In this new version it parses zero or more digits which I think is not what is needed here. Maybe `\d+` is more logical, which is equivalent to `[0-9]+` which is equivalent to `[0-9][0-9]*`.

File: tests/test_libutil.py [hunk 3]
```diff
@@ -165,3 +166,16 @@ def test_extract_keys_recursive(inp, expected):
...
     assert result == expected
+
+
+@pytest.mark.parametrize(
+    "num,expected",
+    [
+        (0, 1),
+        (1, 1),
+        (22, 2),
+        (335, 3),
```
Comment (douweschulte) [inline tests/test_libutil.py:177]: Best to test here are around the border of an extra number so `99` and `100`. As these are the most likely to cause problems.

**Review by douweschulte — CHANGES_REQUESTED** [review]
See my inline comments.

**Review by douweschulte — APPROVED** [review]

File: src/haddock/libs/libutil.py (hunk 1 above)
Comment (joaomcteixeira) [reply src/haddock/libs/libutil.py]: n1!

**Pull request merged**